File: chapter2/lib/mad_lib.ex
defmodule MadLib do
  
  def print_story_with_input() do
    noun = IO.gets "Enter a noun:"
    verb = IO.gets "Enter a verb:"
    adjective = IO.gets "Enter an adjective:"
    adverb = IO.gets "Enter an adverb:"

    noun = replace_trailing_newline(noun)
    verb = replace_trailing_newline(verb)
    adjective = replace_trailing_newline(adjective)
    adverb = replace_trailing_newline(adverb)

    print_story(noun, verb, adjective, adverb)
  end

  def replace_trailing_newline(string) do
    string |> String.replace_trailing("\n", "")
  end

  def print_story(noun, verb, adjective, adverb) do
    "Do you #{verb} your #{adjective} #{noun} #{adverb}? That's hilarious!"
  end
end

File: go-lang/chapter2/madlib/madlib.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)

	fmt.Println("Enter a noun:")
	noun, _ := reader.ReadString('\n')

	fmt.Println("Enter a verb:")
	verb, _ := reader.ReadString('\n')

	fmt.Println("Enter an adjective:")
	adjective, _ := reader.ReadString('\n')

	fmt.Println("Enter an adverb:")
	adverb, _ := reader.ReadString('\n')

	fmt.Printf("Do you %s your %s %s %s? That's hilarious", strings.TrimSuffix(verb, "\n"), strings.TrimSuffix(adjective, "\n"), strings.TrimSuffix(noun, "\n"), strings.TrimSuffix(adverb, "\n"))
}
